core/src/main/java/org/apache/carbondata/core/datamap/DataMapFilter.java

```diff
@@ -100,6 +107,19 @@ public void setTable(CarbonTable table) {
     this.table = table;
   }
 
+  private void initializeExternalSegmentFilter() {
+    if ((expression instanceof AndExpression) && (((AndExpression) expression)
+        .getRight() instanceof InExpression) && (expression.getChildren().get(1).getChildren()
+        .get(0) instanceof ColumnExpression) && (((ColumnExpression) expression.getChildren().get(1)
+        .getChildren().get(0))).getColumnName().equalsIgnoreCase("positionid")) {
```

Comment:
Can use CarbonCommonConstants.CARBON_IMPLICIT_COLUMN_POSITIONID instead of `positionid`

Reply:
done